cupsfilters/ppdgenerator.c

```diff
@@ -1978,47 +1978,93 @@ ppdCreateFromIPP2(char         *buffer,          /* I - Filename buffer */
     is_pdf = 1;
   }
 #ifdef CUPS_RASTER_HAVE_APPLERASTER
-  if (cupsArrayFind(pdl_list, "image/urf")) {
+  else if (cupsArrayFind(pdl_list, "image/urf")) {
```

Comment:
Why "else if" here?

Reply:
I can't take credit for this `else if` as it was in the original cups-filters 1.x pull request https://github.com/OpenPrinting/cups-filters/pull/515

I suspect it is because libppd's  `ppd/ppd-generator.c` has an `else if` there also:
https://github.com/OpenPrinting/libppd/blob/92c64ac08611b7e1ca2e340b6bb0488ea2ee3035/ppd/ppd-generator.c#L697C1-L706

Reply:
libppd's `ppd/ppd-generator.c` does the following :
```C
if (cupsArrayFind(pdl_list, "application/vnd.cups-pdf")) ...
else if (cupsArrayFind(pdl_list, "image/urf")) ...
else if (cupsArrayFind(pdl_list, "application/pdf")) ...
else if (cupsArrayFind(pdl_list, "application/PCLm")) ...
else if (cupsArrayFind(pdl_list, "application/vnd.hp-pclxl")) ...
else if (cupsArrayFind(pdl_list, "application/postscript")) ...
else if (cupsArrayFind(pdl_list, "application/vnd.hp-pcl")) ...
if (cupsArrayFind(pdl_list, "image/jpeg")) ...
if (cupsArrayFind(pdl_list, "image/png")) ...
```

while the existing cups-filters 1.x's cupsfilters/ppdgenerator.c does the following :
```C
if (cupsArrayFind(pdl_list, "application/vnd.cups-pdf")) is_pdf=1 ...
else if (cupsArrayFind(pdl_list, "application/pdf")) is_pdf=1 ...
if (cupsArrayFind(pdl_list, "image/urf")) is_apple=1 ...
if (!is_apple && cupsArrayFind(pdl_list, "image/pwg-raster")) is_pwg=1 ...
if (!is_apple && !is_pwg && cupsArrayFind(pdl_list, "application/PCLm"))  ...
if (!is_pdf && !is_apple && !is_pwg && !is_pclm) {
  if (cupsArrayFind(pdl_list, "application/postscript")) ...
  if (cupsArrayFind(pdl_list, "application/vnd.hp-pcl")) ...
}
if (cupsArrayFind(pdl_list, "image/jpeg")) ...
if (cupsArrayFind(pdl_list, "image/png")) ...
```

So looks like I should replace the `else if` with `if` as the conditional logic is different to libppd's.